src/Expander.hs

```diff
@@ -1025,72 +1027,79 @@ expandLiteralString _dest str =
 -- If we're stuck waiting on a signal, we return that signal and a continuation
 -- in the form of a sequence of closures. The first closure should be applied to
 -- the result of wait-signal, the second to the result of the first, etc.
-interpretMacroAction :: MacroAction -> Expand (Either (Signal, [Closure]) Value)
```

Comment:
mostly whitespace changes:

```
interpretMacroAction (...) = ...
interpretMacroAction (...) = ...
```

to

```
interpretMacroAction prob = \case
  (...) -> ...
  (...) -> ...
```

plus the new `MacroActionWhichProblem` case.